src/audio/host-zephyr.c

```diff
@@ -449,8 +449,20 @@ static int host_copy_normal(struct host_data *hd, struct comp_dev *dev, copy_cal
 	comp_dbg(dev, "host_copy_normal()");
 
 	copy_bytes = host_get_copy_bytes_normal(hd, dev);
-	if (!copy_bytes)
+	if (!copy_bytes) {
+		if (hd->partial_size != 0) {
+			if (stream_sync(hd, dev)) {
+				ret = dma_reload(hd->chan->dma->z_dev, hd->chan->index, 0, 0,
+						 hd->partial_size);
+				if (ret < 0)
+					comp_err(dev, "dma_reload() failed, ret = %u", ret);
+
+				hd->partial_size = 0;
+			}
+		}
```

Comment:
This seems to duplicate the logic at the end of the functions, so how about instaed:

```
if (copy_bytes)
     cb(dev, copy_bytes);

if (!hd->partial_bytes)
       return 0;
```

And let the existing code to call dma_reload at the end of the function.

Reply:
@dnikodem Ack, so above won't work with CONFIG_HOST_DMA_RELOAD_DELAY_ENABLE=y and if we get repeated calls copy_bytes==0 (stream is ending). So back to original version (with just the commit message clarified) is ok to me